src/managers/GitObjectManager.js

```diff
@@ -27,30 +27,24 @@ export class GitObjectManager {
         // Try to get the packfile from the in-memory cache
         let p = PackfileCache.get(filename)
         if (!p) {
-          // If not there, load it from a .idx file
-          const idxName = filename.replace(/pack$/, 'idx')
-          if (await fs.exists(`${gitdir}/objects/pack/${idxName}`)) {
-            const idx = await fs.read(`${gitdir}/objects/pack/${idxName}`)
-            p = await GitPackIndex.fromIdx({ idx, getExternalRefDelta })
-          } else {
-            // If the .idx file isn't available, generate one.
-            const pack = await fs.read(`${gitdir}/objects/pack/${filename}`)
-            p = await GitPackIndex.fromPack({ pack, getExternalRefDelta })
-            // Save .idx file
-            await fs.write(`${gitdir}/objects/pack/${idxName}`, p.toBuffer())
-          }
+          p = GitObjectManager.loadPack(
+            fs,
+            gitdir,
+            filename,
+            getExternalRefDelta
+          )
           PackfileCache.set(filename, p)
         }
         // console.log(p)
         // If the packfile DOES have the oid we're looking for...
+        p = await p
         if (p.offsets.has(oid)) {
           // Make sure the packfile is loaded in memory
           if (!p.pack) {
-            const pack = await fs.read(`${gitdir}/objects/pack/${filename}`)
-            await p.load({ pack })
+            p.pack = fs.read(`${gitdir}/objects/pack/${filename}`)
```

Comment:
Interesting. You've changed `.pack` from a Buffer to a Promise, and moved the await inside of `readSlice` so that the `GitPackIndex` object can be "ready" sooner?

I am OK with this, but if we do this we should also change (or simply delete - I did a quick check and they aren't used anywhere) `GitPackIndex.load` and `GitPackIndex.unload`, because they do not treat `.pack` as a Promise, and that would be confusing.